boot-core/src/tests/unit_tests/daemon_state.rs

```diff
@@ -0,0 +1,49 @@
+/*
+    DaemonState
+*/
+#[cfg(test)]
+mod general {
+    use speculoos::prelude::*;
+
+    use tokio::runtime::Runtime;
+
+    use crate::{
+        daemon::state::{
+            DaemonState,
+            DaemonOptionsBuilder
+        },
+        networks
+    };
+
+    #[test]
+    fn daemon_state() {
+        let rt = Runtime::new().unwrap();
+
+        let network = networks::PISCO_1;
+        let deployment_id = "PISCO_1_TEST_DEPLOY".to_string();
+
+        // build options with DaemonOptionsBuilder
+        let options = DaemonOptionsBuilder::default()
+            .network(network)
+            .deployment_id(deployment_id)
+            .build()
+            .unwrap();
+
+        let daemon_state = DaemonState::new(options);
+
+        // start the daemon
+        let res = rt.block_on(daemon_state).unwrap();
```

Comment:
Since this line reaches out the the PISCO node I don't think it should be in unit test. 

Imo we should set up an integration test that spins up its own daemon is tests the daemon code with that because testnets + endpoints are very unreliable. 

Happy to hear your opinion and maybe we can jump on a call to discuss how we can implement that?

Reply:
totally agree that we should trash this and move the test to an integration test.
after i wrote the unit test i was left thinking that it was pointless but left it for review
so i could get feedback and see how to proceed in this type of occasions